File: istar-backend/src/main/java/com/haui/istar/service/AdminDashboardService.java
package com.haui.istar.service;

import com.haui.istar.dto.dashboard.DashboardStatsDto;

public interface AdminDashboardService {
    /**
     * Lấy dữ liệu thống kê tổng quan và phân tích chuyên sâu cho Dashboard quản trị.
     *
     * @param recruitmentId ID của đợt tuyển (tùy chọn; nếu null sẽ ưu tiên đợt active hoặc toàn bộ)
     * @return DashboardStatsDto chứa các chỉ số, phân bổ ban, phổ trạng thái và xu hướng
     */
    DashboardStatsDto getDashboardStats(Long recruitmentId);
}

File: istar-frontend/src/services/adminDashboardService.ts
import apiClient from "./apiClient";
import { DashboardStats } from "@/types/dashboard";
import { ApiResponse } from "@/types/user";

export const adminDashboardService = {
  /**
   * Lấy dữ liệu thống kê tổng quan và phân tích chuyên sâu cho Dashboard
   * @param recruitmentId Tùy chọn ID đợt tuyển
   */
  getDashboardStats: async (recruitmentId?: number): Promise<DashboardStats> => {
    const params = recruitmentId ? { recruitmentId } : undefined;
    const response = await apiClient.get<ApiResponse<DashboardStats>>(
      "/api/admin/dashboard/stats",
      { params }
    );
    return response.data.data;
  },
};

export default adminDashboardService;
